src/lib/adapters/WebDav.js

```diff
@@ -0,0 +1,400 @@
+            {
+                if (resp.status !== 404)
+                {
+                    throw new Error('Failed to fetch bookmarks :' + resp.status + ":");
+                }
+            }
+        } catch (e) {
+            console.log ("caught error");
+            console.log (e);
+
+            server_db = new Map ();
+        }
+
+        let myBookmarks = Array.from(server_db.values())
+            .map(bm => {
+                    return new Bookmark(bm.id, null, bm.url, bm.title, bm.path)
+                    })
+
+        console.log('Received bookmarks from server')
+        console.log (myBookmarks);
+
+        return myBookmarks
+    }
+
+    async getBookmark (id, autoupdate) {
+        console.log('Fetching single bookmark', this.server);
+        let bm = this.db.get(id);
+        if (!bm) {
+            throw new Error('Failed to fetch bookmark');
+        }
+        let bookmark = new Bookmark(bm.id, null, bm.url, bm.title, bm.path);
+        return bookmark;
+    }
+
+    async createBookmark (bm) {
+        console.log('Create single bookmark', bm, this.server)
+            if (!~['https:', 'http:', 'ftp:'].indexOf(url.parse(bm.url).protocol)) {
+                return false
+            }
+
+        let highestID = 0;
+        this.db.forEach ( (value, key) => {
+            if (value && value.id && highestID < value.id)
+            highestID = value.id;
+        });
+
+        bm.id = highestID + 1;
+
+try {
+        this.db.set(bm.id, {
+            id: bm.id
+            , url: bm.url
+            , title: bm.title
+            , path: bm.path
+        });
+} catch (e) {
+    console.log ("SET FAILED");
+    console.log (e);
+    console.log (this);
+    console.log (bm);
+}
+
+        return bm;
+    }
+
+    async updateBookmark (remoteId, newBm) {
+        console.log('Update bookmark', newBM, remoteId, this.server)
+        if (!~['https:', 'http:', 'ftp:'].indexOf(url.parse(newBm.url).protocol)) {
+            return false
+        }
+        let bm = await this.getBookmark(remoteId, false)
+
+        this.db.set(bm.id, {
+            id: bm.id
+            , url: newBm.url
+            , title: newBm.title
+            , path: newBm.path
+            })
+            console.log ("THIS");
+            console.log (this);
+            return new Bookmark(remoteId, null, newBm.url, newBm.title, newBm.path)
+    }
+
+    async removeBookmark (remoteId) {
+        console.log('Remove bookmark', remoteId, this.server)
+        this.db.delete(remoteId);
+        console.log ("THIS");
+        console.log (this);
+    }
+
+    renderOptions (ctl, rootPath) {
+        let data = this.getData()
+
+        let onchangeURL = (e) => {
+            if (this.saveTimeout) clearTimeout(this.saveTimeout)
+            {
+                let myUrl = e.target.value;
+                let myHttps = 0;
+                if (~['https:'].indexOf(url.parse(myUrl).protocol))
+                    myHttps = 1;
```

Comment:
probably unnecessary, now that you're no longer using the library?

Reply:
right, it was necessary for webdavlib.js that I am no longer using.   I will remove it.